### PR TITLE
Virtualbox

### PR DESCRIPTION
**Pull request by osallou**
Hi,
here is pull request with a few fixes and the virtualbox stuff. I will create a kvm provider with virtio support and make a new pull request

### CODE REVIEW

File: common/tasks/bootstrap.py [hunk 1]
```diff
@@ -13,7 +13,11 @@ def get_bootstrap_args(info):
 		options.append('--include=' + ','.join(include))
 	if len(exclude) > 0:
 		options.append('--exclude=' + ','.join(exclude))
-	arguments = [info.manifest.system['release'], info.root, 'http://http.debian.net/debian']
+        if info.manifest.bootstrapper['mirror']:
+		mirror =  info.manifest.bootstrapper['mirror']
+	else:
+		mirror = 'http://http.debian.net/debian'
```
Comment (andsens) [inline common/tasks/bootstrap.py]: The manifest should decide the defaults, not the tasks. You can see how it is done with the tarball location in the ec2 manifest.

File: providers/virtualbox/tasks/boot.py [hunk 2]
```diff
@@ -7,19 +7,29 @@ class ConfigureGrub(Task):
...
+        	script_dst = os.path.join(info.root, 'etc/grub.d/10_linux')
+        	copy(script_src, script_dst)
+        	os.chmod(script_dst, rwxr_xr_x)
+                script_src = os.path.normpath(os.path.join(os.path.dirname(__file__), '../assets/grub.d/00_header'))
+                script_dst = os.path.join(info.root, 'etc/grub.d/00_header')
+                copy(script_src, script_dst)
+                os.chmod(script_dst, rwxr_xr_x)
+		log_check_call(['/usr/sbin/chroot', info.root, 'update-initramfs', '-u'])
+		# Install grub in mbr
+		log_check_call(['grub-install', '--boot-directory='+info.root+"/boot/", info.bootstrap_device['path']])
```
Comment (andsens) [inline providers/virtualbox/tasks/boot.py]: Remember to use absolute paths when calling stuff, we shouldn't rely on $PATH.

File: providers/virtualbox/tasks/packages.py [hunk 3]
```diff
@@ -32,7 +32,8 @@ def run(self, info):
 		               # isc-dhcp-client doesn't work properly with ec2
 		               'dhcpcd',
 		               'chkconfig',
-		               'openssh-client'
+		               'openssh-client',
+			       'grub2'
```
Comment (andsens) [inline providers/virtualbox/tasks/packages.py]: Indentation seems a bit wrong here. I use tabs to indent, I know that differs from PEP8 and maybe we should conform to that, but lets just keep it consistent right now and use tabs for indentation and spaces for alignment.

File: providers/virtualbox/tasks/boot.py [hunk 4]
```diff
@@ -7,19 +7,29 @@ class ConfigureGrub(Task):
...
-			                          root_path=info.bootstrap_device['partitions']['root_path'])))
+			device_map.write('(hd0)   /dev/sda\n')
 
 		from common.tools import log_check_call
-		log_check_call(['/usr/sbin/chroot', info.root, 'update-initramfs', '-u'])
 
-		log_check_call(['/usr/sbin/chroot', info.root,
-		                '/usr/sbin/grub-mkconfig', '-o', '/boot/grub/grub.cfg'])
+		from shutil import copy
+        	script_src = os.path.normpath(os.path.join(os.path.dirname(__file__), '../assets/grub.d/10_linux'))
```
Comment (andsens) [inline providers/virtualbox/tasks/boot.py:23]: Why do we need this when installing grub2? The menu.lst is meant to be there for the old paravirtualized grub bootloader. But virtualbox loads the grub2 we installed. Are you sure it doesn't work without this?

Comment (osallou) [inline providers/virtualbox/tasks/boot.py:23]: 2013/8/13 Anders Ingemann notifications@github.com

> In providers/virtualbox/tasks/boot.py:
> 
> > -       log_check_call(['/usr/sbin/chroot', info.root,
> > -                       '/usr/sbin/grub-mkconfig', '-o', '/boot/grub/grub.cfg'])
> > -       from shutil import copy
> > -           script_src = os.path.normpath(os.path.join(os.path.dirname(**file**), '../assets/grub.d/10_linux'))
> 
> Why do we need this when installing grub2? The menu.lst is meant to be
> there for the old paravirtualized grub bootloader. But virtualbox loads the
> grub2 we installed. Are you sure it doesn't work without this?
> 
> The generated grub.cfg from grub.d contains loopback references in menu
> entries. We need to override some "defaults" to put our own menu
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/andsens/build-debian-cloud/pull/86/files#r5730603
> .

## 

gpg key id: 4096R/326D8438  (keyring.debian.org)

Key fingerprint = 5FB4 6F83 D3B9 5204 6335  D26D 78DC 68DB 326D 8438

**Pull request merged**